pkg/fileutils/fileutils_test.go

```diff
@@ -1,10 +1,94 @@
 package fileutils
 
 import (
+	"io/ioutil"
 	"os"
+	"path"
 	"testing"
 )
 
+// CopyFile with invalid src
+func TestCopyFileWithInvalidSrc(t *testing.T) {
+	tempFolder, err := ioutil.TempDir("", "docker-fileutils-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	bytes, err := CopyFile("/invalid/file/path", path.Join(tempFolder, "dest"))
+	if err == nil {
+		t.Fatal("Should have fail to copy an invalid src file")
+	}
+	if bytes != 0 {
+		t.Fatal("Should have written 0 bytes")
+	}
+
+}
+
+// CopyFile with invalid dest
+func TestCopyFileWithInvalidDest(t *testing.T) {
+	tempFolder, err := ioutil.TempDir("", "docker-fileutils-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	src := path.Join(tempFolder, "file")
+	_, err = os.Create(src)
```

Comment:
I would put something in the file so that its not "zero size" then when you check "bytes != 0" you can be sure that its zero because it didn't copy it and not because it copied an zero size file.